test/object-store/c_api/c_api.cpp

```diff
@@ -2460,6 +2460,18 @@ TEST_CASE("C API", "[c_api]") {
                         CHECK(rlm_stdstr(a2) == "a");
                         CHECK(rlm_stdstr(b2) == "b");
                         CHECK(c2.type == RLM_TYPE_NULL);
+
+                        size_t out_index = -1;
+                        bool found;
+                        CHECK(checked(realm_list_find(strings.get(), &a2, &out_index, &found)));
```

Comment:
These tests only seem to test the case where a value is found. Can we add a test where `find` is called with a value that doesn't exist in the list?

Reply:
This was actually a test that was exercising the "not found" path, but we are sneaking in Mixed Links support as well. Let me add a test.